### ISSUE TITLE
Error with creating resource `security group rules` when `create_default_security_group` is `false`

### ISSUE THREAD

**Issue by s2504s** [cloudposse/terraform-aws-ec2-instance#18]
### Terraform Version
Terraform v0.10.8

### Output
```
Error running plan: 1 error(s) occurred:

* module.vm.aws_security_group_rule.egress: 1 error(s) occurred:

* module.vm.aws_security_group_rule.egress: Resource 'aws_security_group.default' not found for variable 'aws_security_group.default.id'
```
### Terraform Configuration Files
```terraform

module "vm" {
  source                        = "git::https://github.com/cloudposse/terraform-aws-ec2-instance.git?ref=tags/0.5.0"
  associate_public_ip_address   = "false"
  create_default_security_group = "false"
  source_dest_check             = "false"
  name                          = "vm"
  namespace                     = "${var.namespace}"
  stage                         = "${var.stage}"
  vpc_id                        = "${module.vpc.vpc_id}"
  security_groups               = ["${aws_security_group.vm.id}"]
}
```
### Debug Output

```terraform
------------------------------------------------------------------------
Error running plan: 1 error(s) occurred:

* module.vm.aws_security_group_rule.egress: 1 error(s) occurred:

* module.vm.aws_security_group_rule.egress: Resource 'aws_security_group.default' not found for variable 'aws_security_group.default.id'

```

### Expected Behavior
Instance should be created

### Actual Behavior
```
Error running plan: 1 error(s) occurred:
```
### Steps to Reproduce
* `terraform plan`

**Comment by s2504s** [comment]
![image](https://user-images.githubusercontent.com/1134449/33165687-67f74494-d040-11e7-80cb-48f84416e4cb.png)

**Comment by comeanother** [comment]
Resolved in #19
